### ISSUE TITLE
TypeSerializer's do not work

### ISSUE THREAD

**Issue by jtaekema** [pardom-zz/ActiveAndroid#8]
The deserialize method always gets a null object passed into it.

**Comment by dbalaouras** [comment]
Which TypeSerializer is that? It's always helpful to give an example and/or steps to reproduce.

**Comment by zach-klippenstein** [comment]
I can confirm this, using a custom TypeSerializer:

``` java
public class InetAddressSerializer extends TypeSerializer
{
  @Override
  public Object deserialize(Object data)
  {
    if (data == null)
    {
      return null;
    }

    try
    {
      return InetAddress.getByAddress((byte[])data);
    }
    catch (UnknownHostException e)
    {
      throw new RuntimeException("Host corrupted in database", e);
    }
  }

  @Override
  public Class<?> getDeserializedType()
  {
    return InetAddress.class;
  }

  @Override
  public SQLiteType getSerializedType()
  {
    return SQLiteType.BLOB;
  }

  @Override
  public Object serialize(Object data)
  {
    if (data == null)
    {
      return null;
    }

    return ((InetAddress)data).getAddress();
  }
}
```

What's happening is that the if-chain in Model.loadFromCursor() is pulling the value from the Cursor based on the fieldType (in this case InetAddress), but it only checks primitive types and Models. Model.save() doesn't have this problem because it pulls the serialized class from the object returned by serialize (byte[] in this case). It seems like what should be happening is the load code should read the cursor type based on the serializer's SQL type if there's a serializer, and check for primitive types and models if there's no serializer.

I'm working on and testing a fix for this.

**Comment by dbalaouras** [comment]
I can confirm this is indeed an issue..good catch!

**Comment by pardom-zz** [comment]
Fixed in 8a747a2.
